src/utils/telemetry.ts

```diff
@@ -193,13 +239,41 @@ export class DefaultTelemetryProvider implements TelemetryProvider {
       this.exporter.sendCount({
         name,
         timestamp: new Date().toISOString(),
-        attributes,
+        attributes: this.insertDefaultAttributes(attributes),
         value,
       });
     } catch (err) {
       Logger.getInstance().debug(`Error sending telemetry count: ${err.message}`);
     }
   }
+
+  /**
+   * Configures default attributes which are added as metadata to all collected
+   * collected telemetry events
+   *
+   * @param attributes The default attributes to include with all telemetry events
+   */
+  withDefaultAttributes(attributes: Attributes): DefaultTelemetryProvider {
+    this.defaultAttributes = attributes;
+    return this;
+  }
+
+  private insertDefaultAttributes(attributes?: Attributes): Attributes {
+    // We're avoiding Object.assign and object literal spreading since they're
+    // not supported by IE11
+    const mergedAttributes: Attributes = {};
+    [attributes, this.defaultAttributes].forEach((attributeSet) => {
```

Comment:
I wonder if you could do something like:

```
const mergedAttributes = this.defaultAttributes
for (const [key, value] of Object.entries(attributes)) {
  mergedAttributes[key] = value;
}
return mergedAttribute;
```

Reply:
This or object literal spreading would be a lot simpler. I've been trying to avoid APIs not supported by IE11 since we still have some open work to remove a few lingering IE11 errors in DLO. If we decide to no longer support IE11 in DLO or add something like babel to build process we could use `Object.assign` or object literal spreading.

Reply:
I see, I just straight missed this comment:

```
// We're avoiding Object.assign and object literal spreading since they're
// not supported by IE11
```

Reply:
I'm glad you asked about this. I learned we can use object literal spreading and TypeScript (based on our compiler options) will use and polyfill `Object.assign`. We can simplify this function to:

```
  private insertDefaultAttributes(attributes?: Attributes): Attributes {
    return { ...this.defaultAttributes, ...attributes };
  }
```

When built, this results in:

```
    DefaultTelemetryProvider.prototype.insertDefaultAttributes = function (attributes) {
        return __assign(__assign({}, this.defaultAttributes), attributes);
    };

    ...

    var __assign = function() {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };
```

Minified build size is roughly the same, though full build size is ~1KB smaller.